File: Data_Structures/python/09_걸그룹마스터준석이.py
# fast campus 강의
# https://www.acmicpc.net/problem/16165
# 2

N, M = map(int, input().split())
team_mem, mem_team = {}, {}

for i in range(N):
    group_name, member_num = input(), int(input())
    team_mem[group_name] = []
    for j in range(member_num):
        member = input()
        team_mem[group_name].append(member)
        mem_team[member] = group_name

for i in range(M):
    exam_name, exam_num = input(), int(input())
    if exam_num:
        print(mem_team[exam_name])
    else:
        for member in sorted(team_mem[exam_name]):
            print(member)

File: Data_Structures/c++/09_걸그룹마스터준석이.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/16165
// 1

#include <iostream>
#include <map>
#include <set>

using namespace std;

int main()
{
	int N, M;
	cin >> N >> M;
	map<string, string> map_groups;		// key:member, value:group
	map<string, set<string>> map_members;	// key:group, value:members

	for (int i = 0; i < N; i++) {
		string group;
		int cnt;
		cin >> group >> cnt;

		for (int j = 0; j < cnt; j++) {
			string member;
			cin >> member;
			map_members[group].insert(member);
			map_groups[member] = group;
		}
	}

	for (int i = 0; i < M; i++) {
		int cnt;
		string q;
		cin >> q >> cnt;

		if (cnt == 0) {
			for (auto m : map_members[q])
				cout << m << '\n';
		}
		else cout << map_groups[q] << '\n';
	}
}